### PR TITLE
Set default output directory based on os

### PR DESCRIPTION
**Pull request by JCFarrow**
Also add a reset to default button

user/Videos/Bible Karaoke Videos for widows 10/8 + linux
user/My Videos/BKV for windows 7
user/BKV for mac

image

### CODE REVIEW

File: src/App/components/Settings.tsx [hunk 1]
```diff
@@ -119,16 +120,22 @@ export default function Settings(): JSX.Element {
...
+            buttonText="Save videos to..."
+            file={settings.outputDirectory}
+            options={{
+              title: 'Select Output Folder',
+              properties: ['openDirectory'],
+            }}
+            onFileSelected={settings.setOutputDirectory}
+          />
+          <Tooltip content="Reset to default directory">
+              <Button minimal icon="reset" onClick={(): Function => settings.setOutputDirectory(DEFAULT_OUTPUT_DIRECTORY)} />
```
Comment (irahopkinson) [inline src/App/components/Settings.tsx:135]: I'd declare this arrow function as a `const` variable above and refer to the variable here. Naming the function makes it more readable.

**Review by irahopkinson — CHANGES_REQUESTED** [review]
Looks good. I have one suggested change but you can argue about it if you like.

File: src/App/components/Settings.tsx (hunk 1 above)
Comment (JCFarrow) [reply src/App/components/Settings.tsx]: Done. I also positioned the button's tooltip underneath.

File: src/App/components/Settings.tsx [hunk 2]
```diff
@@ -94,6 +95,7 @@ export default function Settings(): JSX.Element {
   const { settings } = useStores();
   const { analytics } = useAnalytics();
   const repoUrl = repository.url.replace(/\.git$/, '');
+  const resetOutputDir = (): Function => settings.setOutputDirectory(DEFAULT_OUTPUT_DIRECTORY);
```
Comment (irahopkinson) [inline src/App/components/Settings.tsx:98]: The actual return type of this arrow function should be `void `, i.e. it doesn't return anything (and certainly doesn't return a `Function`).

**Review by irahopkinson — CHANGES_REQUESTED** [review]
Thanks for fixing the tooltip position as well. Nearly there.

File: src/App/components/Settings.tsx (hunk 2 above)
Comment (JCFarrow) [reply src/App/components/Settings.tsx]: Ok, done

**Review by irahopkinson — APPROVED** [review]
👍 LGTM

**Pull request merged**